tests/unit_tests/test_pychop.py

```diff
@@ -81,6 +81,20 @@ def matrix_nonfermi_id(matrix_row: tuple[int, int, int]) -> str:
         )
 )
 
+INSTRUMENTS_V2 = [[('ARCS', 'ARCS')], [('SEQUOIA', 'SEQUOIA')]]
+INSTRUMENT_CONFIGURATIONS_V2 = [
+    ['SEQ-100-2.0-AST', 'SEQ-700-3.5-AST', 'ARCS-100-1.5-AST', 'ARCS-700-1.5-AST',
+     'ARCS-700-0.5-AST', 'ARCS-100-1.5-SMI', 'ARCS-700-1.5-SMI'],
+    ['High-Resolution', 'High-Flux', 'SEQ-100-2.0-AST', 'SEQ-700-3.5-AST', 'ARCS-100-1.5-AST',
+     'ARCS-700-1.5-AST', 'ARCS-700-0.5-AST', 'ARCS-100-1.5-SMI', 'ARCS-700-1.5-SMI'],
+]
+INSTRUMENT_MATRIX_FERMI_V2 = list(
+        itertools.chain.from_iterable(
+            itertools.product(instr, configurations)
+                  for instr, configurations in zip(INSTRUMENTS_V2, INSTRUMENT_CONFIGURATIONS_V2)
+        )
+)
```

Comment:
What is the logic here?
```
INSTRUMENT_MATRIX_FERMI_V2 = [
    pair for instr, configurations in zip(...)) for pair in product(instr, configurations)
]
```
?
Seems less magical.

Reply:
@ajjackson is better poised to answer that

Reply:
I'm not a big fan of double list comprehensions, I find them hard to read and really only use them for performance.

As Jacob has written it I _can_ tell that the second `for` is the "inner" loop because that's the only way it could work. It's not so bad; up to you @RastislavTuranyi

Reply:
Nor am I, to be honest, it's one of the things I miss from other languages like e.g. Haskell
```haskell
cartProd :: [a] -> [b] -> [(a, b)]
cartProd xs ys = [(x, y) | x <- xs, y <- ys]

INSTRUMENT_MATRIX_FERMI_V2 = [pair | (a, b) <- zip INSTRUMENTS_V2 INSTRUMENT_CONFIGURATIONS_V2,
                                     pair <- cartProd a b]
```
Or even Raku with hyperoperators
```raku
$INSTRUMENT_MATRIX_FERMI_V2 = $INSTRUMENTS_V2 «X» $INSTRUMENT_CONFIGURATIONS_V2
```
Where:
- `X` is cart prod
-  `«f()»` applies the function drawing from each list, and `»` point to the shorter list (i.e. this says they should be of each length)

Reply:
I personally don't mind either way, so might as well leave it for now